plugins/modules/ec2_key.py

```diff
@@ -287,23 +322,34 @@ def main():
         state=dict(default='present', choices=['present', 'absent']),
         tags=dict(type='dict'),
         purge_tags=dict(type='bool', default=False),
+        key_type=dict(type='str', choices=['rsa', 'ed25519']),
         wait=dict(type='bool', removed_at_date='2022-06-01', removed_from_collection='amazon.aws'),
         wait_timeout=dict(type='int', removed_at_date='2022-06-01', removed_from_collection='amazon.aws')
     )
 
-    module = AnsibleAWSModule(argument_spec=argument_spec, supports_check_mode=True)
+    module = AnsibleAWSModule(
+        argument_spec=argument_spec,
+        mutually_exclusive=[
+            ['key_material', 'key_type']
+        ],
```

Comment:
Does Amazon not support providing your own ED25519 key?

Reply:
yes it does, but when providing your own key the type parameter is useless as the provided is either RSA or ED25519 depending on the syntax